plugins/transport/socket/main.go

```diff
@@ -81,6 +81,7 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	go func() {
 		for {
 			n, err := pc.Read(msgBuffer)
+			msgBuffer[n] = 0
```

Comment:
Why did you need to do this?

Reply:
I just tried without this line again. I'm getting `invalid character 'd' after top-level value` when I receive a message that is shorter than the previous one and try to unmarshal it. So I add a 0 into the buffer right after the received message. Maybe I'm not unmarshaling it in the handler correctly?

Reply:
oh wow I think you spotted a pretty critical bug. Prior to this the write statement was w(msgBuffer[:n]) for that very reason, but somehow it has been switched to w(msgBuffer). That's probably on me